llumnix/backends/vllm/worker.py

```diff
@@ -11,51 +11,69 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import List
+from typing import Dict, List
+import math
+import os
 import ray
 import torch
 
 from ray.util.scheduling_strategies import PlacementGroupSchedulingStrategy, NodeAffinitySchedulingStrategy
 from vllm.utils import is_pin_memory_available
 from vllm.worker.worker import Worker
+from vllm.config import CacheConfig, ModelConfig, ParallelConfig
+from vllm.worker.cache_engine import CacheEngine
 
 from llumnix.logger import init_logger
 from llumnix.backends.vllm.utils import _sample_with_torch
 
+from llumnix.backends.vllm.migrate_backend import get_migrate_backend
+from llumnix.config import MigrationConfig
+
 logger = init_logger(__name__)
 NUMPY_SUPPORT_DTYPES = [torch.float32, torch.float16]
 
-@ray.remote(num_cpus=0)
-class RecvActor:
-    def recv_cpu_cache(self, src_worker_handle, src_blocks, is_driver_worker):
-        """
-        Args:
-            src_worker_handle: src worker actor handle
-            blocks: block to send
-        """
-        try:
-            if is_driver_worker:
-                migration_cache = ray.get(src_worker_handle.execute_engine_method.remote("send_cpu_cache", src_blocks))
-            else:
-                migration_cache = ray.get(src_worker_handle.execute_method.remote("send_cpu_cache", src_blocks))
-        # pylint: disable=try-except-raise
-        except:
-            raise
-        return migration_cache
-
 class MigrationWorker(Worker):
     def __init__(self, *args, **kwargs) -> None:
         # replace sampler
         # pylint: disable=import-outside-toplevel
         import vllm.model_executor.layers.sampler
         vllm.model_executor.layers.sampler._sample_with_torch = _sample_with_torch
+
+        backend = os.environ.get("MIGRATE_BACKEND", "rpc")
```

Comment:
Why not pass 'MIGRATE_BACKEND' and 'MIGRATE_CACHE_SIZE' through migration_config rather than environment variables?